### PR TITLE
SEG-25

### PR DESCRIPTION
**Pull request by vjnathe-webonise**
**What does this PR do?**
Tracks the **setSiteType** event based on the device: "m FOR MOBILE OR t FOR TABLET OR d FOR DESKTOP"

**Are there breaking changes in this PR?**
No.

**Any background context you want to provide?**
Earlier initialise of criteo was not setting device type.

**Is there parity with the server-side/android/iOS integration components (if applicable)?**
Yes, difference in device type. 

**Does this require a new integration setting? If so, please explain how the new setting works**
No.

**Links to helpful docs and other external resources**
https://support.criteo.com/s/article?article=202806891-Criteo-OneTag-on-your-homepage&language=en_US

### CODE REVIEW

File: integrations/criteo/lib/index.js [hunk 1]
```diff
@@ -253,3 +254,23 @@ function getProductMetadata(track) {
...
+ * Get Device Type
+ *
+ * @api private
+ *
+ * REF: https://support.criteo.com/s/article?article=202806931-Managing-your-different-site-types&language=en_US
+ */
+
+function getDeviceType() {
+  var ipadRegex = /iPad/;
+  var mobileRegex = /Mobile|iP(hone|od)|Android|BlackBerry|IEMobile|Silk/;
```
Comment (ameyawebonise) [inline integrations/criteo/lib/index.js]: should it not be `iP(hone|ad)` for iPhone or iPad?Are we catering to users agents on the iPod?

Comment (ameyawebonise) [reply integrations/criteo/lib/index.js]: nvm we have the iPad regex separately.but iPod?

Comment (vjnathe-webonise) [reply integrations/criteo/lib/index.js]: This regex referred from criteo's docs only.. see the reference link

Comment (ameyawebonise) [reply integrations/criteo/lib/index.js]: gotcha.

**Review by ameyawebonise — APPROVED** [review]

File: integrations/criteo/lib/index.js (hunk 1 above)
Comment (CarlosMecha) [reply integrations/criteo/lib/index.js]: Could we declare these regex at the beginning of this file?

File: integrations/criteo/lib/index.js [hunk 2]
```diff
@@ -253,3 +254,23 @@ function getProductMetadata(track) {
...
+  if (ipadRegex.test(navigator.userAgent)) {
```
Comment (CarlosMecha) [inline integrations/criteo/lib/index.js:276]: let's make sure `navigator` exists and has a userAgent before we use it.

File: integrations/criteo/test/index.test.js [hunk 3]
```diff
@@ -62,6 +62,41 @@ describe('Criteo', function() {
...
+      it('should track setSiteType for desktop', function() {
+        analytics.stub(window.criteo_q, 'push');
+        analytics.initialize();
+        analytics.called(window.criteo_q.push, {
+          event: 'setSiteType',
+          type: 'd'
+        });
+      });
+
+      it('should track setSiteType for mobile', function() {
```
Comment (CarlosMecha) [inline integrations/criteo/test/index.test.js:75]: Let's also create a test when userAgent is something unexpected (other than empty).

**Review by CarlosMecha — CHANGES_REQUESTED** [review]
Great job! Left a couple of comments. I think the title of the PR should be `SEG-25` too :)

File: integrations/criteo/lib/index.js (hunk 2 above)
Comment (vjnathe-webonise) [reply integrations/criteo/lib/index.js:276]: what needs to be return if it doesn't exist ? `d` that is desktop ?

**Comment by vjnathe-webonise** [comment]
@CarlosMecha Resolved the PR comments.

File: integrations/criteo/lib/index.js (hunk 2 above)
Comment (CarlosMecha) [reply integrations/criteo/lib/index.js:276]: Yep!

**Review by CarlosMecha — APPROVED** [review]
:rocket:

**Comment by vjnathe-webonise** [comment]
Closing this PR as it's merged to upstream.
